cpp/src/strings/combine.cu

```diff
@@ -0,0 +1,305 @@
+    auto d_results_chars = chars_view.data<char>();
+    thrust::for_each_n(execpol->on(stream), thrust::make_counting_iterator<size_type>(0), strings_count,
+        [d_columns, num_columns, d_separator, d_narep, d_results_offsets, d_results_chars] __device__(size_type idx){
+            bool null_element = thrust::any_of( thrust::seq, d_columns, d_columns+num_columns,
+                    [idx] (column_device_view col) { return col.is_null(idx);});
+            if( null_element && d_narep.is_null() )
+                return; // do not write to buffer at all if any element is null
+            size_type offset = d_results_offsets[idx];
+            char* d_buffer = d_results_chars + offset;
+            for( size_type col_idx=0; col_idx < num_columns; ++col_idx )
+            {
+                auto d_column = d_columns[col_idx];
+                if( d_column.nullable() && d_column.is_null(idx) )
+                    d_buffer = detail::copy_string(d_buffer, d_narep);
+                else
+                {
+                    string_view d_str = d_column.element<string_view>(idx);
+                    d_buffer = detail::copy_string(d_buffer, d_str);
+                }
+                // separator only in between elements
+                if( col_idx+1 < num_columns )
+                    d_buffer = detail::copy_string(d_buffer, d_separator);
+            }
+        });
+
+    return make_strings_column(strings_count, std::move(offsets_column), std::move(chars_column),
+                               null_count, std::move(null_mask), stream, mr);
+}
+
+//
+std::unique_ptr<cudf::column> join_strings( strings_column_view strings,
+                                            const std::string separator = "",
+                                            const char* narep=nullptr,
+                                            rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                            cudaStream_t stream=0 )
+{
+    auto strings_count = strings.size();
+    if( strings_count == 0 )
+        return detail::make_empty_strings_column(mr,stream);
+
+    auto execpol = rmm::exec_policy(stream);
+    auto separator_ptr = detail::string_from_host(separator.c_str(), stream);
+    auto d_separator = *separator_ptr;
+    auto narep_ptr = detail::string_from_host(narep, stream);
+    string_view d_narep(nullptr,0);
+    if( narep_ptr )
+        d_narep = *narep_ptr;
+
+    auto strings_column = column_device_view::create(strings.parent(),stream);
+    auto d_column = *strings_column;
+    auto d_offsets = strings.offsets().data<int32_t>();
+
+    // create an offsets array for building the output memory layout
+    rmm::device_vector<size_type> output_offsets(strings_count+1);
+    auto d_output_offsets = output_offsets.data().get();
+    // using inclusive-scan to compute last entry which is the total size
+    thrust::transform_inclusive_scan( execpol->on(stream),
```

Comment:
This looks like it can just use `detail::make_offsets_child_column`

Reply:
Not trying to build a column here. This is just a set of temporary offsets that are needed for a later kernel.